automation/taskcluster/lib/tasks.py

```diff
@@ -150,48 +174,133 @@ def craft_beetmover_task(
+            ),
+            DOCKER_IMAGE_ROUTE_LATEST_PATTERN.format(
+                trust_level=self.trust_level, image_name=image_name
+            ),
+            DOCKER_IMAGE_ROUTE_PUSH_DATE_PATTERN.format(
+                trust_level=self.trust_level, image_name=image_name,
+                year=self.push_date_time.year,
+                month=self.push_date_time.month,
+                day=self.push_date_time.day,
+                build_id=self.push_date_time.strftime('%Y%m%d%H%M%S'),
+            ),
+        ]
+
+        return self._craft_default_task_definition(
+            'mobile-{}-images'.format(self.trust_level),
+            'aws-provisioner-v1',
+            dependencies=[],
+            routes=routes,
+            scopes=[],
+            name='Android Components - Build docker image "{}"'.format(image_name),
+            description='',
+            payload={
+                # This sha points to the tag docker:1.6-dind. 1.6 is what's run on taskcluter.
+                'image': 'docker@sha256:db593745f76eeb9a58b8679e85bf8651b8efc0e9ff1132fb75bc613e24e16bfc',
+                'maxRunTime': 7200,     # The build docker image takes more than an hour to complete
+                'command': [
+                    '/bin/sh',
+                    '--login',
+                    '-xec',
+                    (
+                        # Needed to install zstd
+                        "sed -i -e 's/v[[:digit:]]\.[[:digit:]]/edge/g' /etc/apk/repositories"
+                        " && apk add --update zstd git"
+                        " && git clone {repo_url} repo"
+                        " && cd repo"
+                        " && git config advice.detachedHead false"
+                        " && git checkout {commit}"
+                        " && cd automation/docker/{image_folder}"
+                        " && docker build -t taskcluster-built ."
+                        " && docker save taskcluster-built | zstd > /image.tar.zst".format(
+                            repo_url=self.repo_url,
+                            commit=self.commit,
+                            image_folder=image_name
+                        )
+                    )
+                ],
+                'features': {
+                    'chainOfTrust': True,   # Needed to ensure the Docker image is trusted
+                    'dind': True,
+                },
+                'artifacts': {
+                    'public/image.tar.zst': {
+                        'type': 'file',
+                        'expires': taskcluster.stringDate(taskcluster.fromNow(DEFAULT_EXPIRES_IN)),
+                        'path': '/image.tar.zst',
+                  }
+                }
+            },
+            extra={
+                'index': {
+                    # This automatically rebuilds the docker images once a month.
+                    'expires': taskcluster.stringDate(taskcluster.fromNow('1 month')),
```

Comment:
I understand the desire for this, but I'm a little leary of causing the image to automatically disappear. If there become issues rebuilding the image during a chemspill, for example. I wonder if it would be possible to look at the task time when looking up the image in the index, and rebuilding if it is too old. That way, it is easy enough to remove that check and proceed with the release.